grype/presenter/table/presenter.go

```diff
@@ -14,19 +14,27 @@ import (
 	"github.com/anchore/grype/grype/vulnerability"
 )
 
+const (
+	AppendSuppressed = "-(suppressed)"
+)
+
 // Presenter is a generic struct for holding fields needed for reporting
 type Presenter struct {
 	results          match.Matches
+	ignoredMatches   []match.IgnoredMatch
 	packages         []pkg.Package
 	metadataProvider vulnerability.MetadataProvider
+	inclSuppressed   bool
 }
 
 // NewPresenter is a *Presenter constructor
-func NewPresenter(results match.Matches, packages []pkg.Package, metadataProvider vulnerability.MetadataProvider) *Presenter {
+func NewPresenter(results match.Matches, ignoredMatches []match.IgnoredMatch, packages []pkg.Package, metadataProvider vulnerability.MetadataProvider, inclSuppressed bool) *Presenter {
```

Comment:
nit: we should probably leave `results match.Matches, packages []pkg.Package, metadataProvider vulnerability.MetadataProvider` as the first 3 parameters, since the other 2 are related and would make a bit more sense to be grouped together.

I wonder if we could simplify this a little so we just pass in `ignoredMatches []match.IgnoredMatch` if `presenterConfig.includeSuppressed` is true so we don't need to track the boolean to display them in multiple places. It would change [this line](https://github.com/anchore/grype/pull/966/files#diff-e12805a26e8031bd6f10e235003454ad4ebd52a6af6c78c740df26193e41d28dR30) to something like:

```go
if presenterConfig.includeSuppressed {
    return table.NewPresenter(matches, packages, metadataProvider, ignoredMatches)
}
return table.NewPresenter(matches, packages, metadataProvider, nil)
```